server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/BaseStarTreeBuilder.java

```diff
@@ -133,10 +134,11 @@ public List<MetricAggregatorInfo> generateMetricAggregatorInfos(MapperService ma
                 if (metricStat.isDerivedMetric()) {
                     continue;
                 }
-                IndexNumericFieldData.NumericType numericType;
+                FieldValueConverter fieldValueConverter;
                 Mapper fieldMapper = mapperService.documentMapper().mappers().getMapper(metric.getField());
-                if (fieldMapper instanceof NumberFieldMapper) {
-                    numericType = ((NumberFieldMapper) fieldMapper).fieldType().numericType();
+                if (fieldMapper instanceof ParametrizedFieldMapper
```

Comment:
why ParametrizedFieldMapper ? can we check for type of the base class rather. 
DocCountFieldMapper for instance is of type `MetadataFieldMapper`

Reply:
instance of `FieldMapper` should work.